packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -593,7 +611,16 @@ export function authorizationServiceBuilder(
       assertOrganizationIsClientConsumer(organizationId, client.data);
 
       const purpose = await retrievePurpose(purposeId, readModelService);
-      assertOrganizationIsPurposeConsumer(organizationId, purpose);
+      const delegationId = purpose.delegationId;
+      const delegation = delegationId
+        ? await retrieveActiveConsumerDelegation(delegationId, readModelService)
+        : undefined;
```

Comment:
For consistency, since it's used only for this need, you could move this check into the retrieve and call it retrievePurposeDelegation as done here: https://github.com/pagopa/interop-be-monorepo/pull/1407/files#diff-95b4e078c342343517e86a98ea5650ede6c4ec6e5c9f4547dbfb44c8d6d22118R229

As you prefer, no strong opinion

Reply:
If you do that, the error also could become a more specific `purposeDelegationNotFound` (see same PR)